static/src/js/components/OrderProgressList/__tests__/OrderProgressItem.test.js

```diff
@@ -1,57 +1,121 @@
 import React from 'react'
-import Enzyme, { shallow } from 'enzyme'
-import Adapter from '@wojtekmaj/enzyme-adapter-react-17'
 
-import { Badge, ProgressBar } from 'react-bootstrap'
+import { render, screen } from '@testing-library/react'
+
 import { OrderProgressItem } from '../OrderProgressItem'
 
-import { retrievalStatusPropsEsi } from '../../OrderStatus/__tests__/mocks'
+import {
+  retrievalStatusPropsEsi,
+  retrievalStatusPropsSwotOrder,
+  retrievalStatusPropsHarmonyOrder
+} from '../../OrderStatus/__tests__/mocks'
 
 beforeEach(() => {
   jest.clearAllMocks()
 })
 
-Enzyme.configure({ adapter: new Adapter() })
+const setup = (type, overrideProps) => {
+  let orderOrigin = null
+  let props = {}
+
+  switch (type) {
+    case 'SWODLR':
+      orderOrigin = retrievalStatusPropsSwotOrder
+      break
+    case 'ESI':
+      orderOrigin = retrievalStatusPropsEsi
+      break
+    case 'Harmony':
+      orderOrigin = retrievalStatusPropsHarmonyOrder
+      break
+    default:
+      orderOrigin = retrievalStatusPropsEsi
+      break
+  }
 
-function setup(overrideProps) {
-  const order = retrievalStatusPropsEsi.retrieval.collections.byId[1].orders[0]
+  const { retrieval } = orderOrigin
+  const { collections } = retrieval
+  const { byId } = collections
+  const { orders } = byId[1]
 
-  const props = {
+  const order = orders[0]
+
+  props = {
     order,
     ...overrideProps
   }
 
-  const enzymeWrapper = shallow(<OrderProgressItem {...props} />)
+  render(<OrderProgressItem {...props} />)
 
   return {
-    enzymeWrapper,
     props
   }
 }
 
 describe('OrderProgressItem component', () => {
-  test('shows the correct order metadata', () => {
-    const { enzymeWrapper } = setup()
+  describe('Complete Swodlr Order', () => {
+    test('shows the correct order metadata', () => {
+      setup('SWODLR')
+
+      expect(screen.getByRole('heading', {
+        level: 5,
+        name: 'Order ID: e7efe743-f253-43e3-b017-74faa8bdfcf1'
+      })).toBeInTheDocument()
+
+      expect(screen.getByText('Complete')).toBeInTheDocument()
```

Comment:
We need better assertions than just 'Complete' shows up on the page. Where does it show up on the page, or under what context does it show up on the page?

Reply:
@bnp26 One thing you can do is get the element and then do something like `element.parentElement` and use `within`

Reply:
That's bad practice to do that. React testing library doesn't like it when you use DOM like that. Better to add the appropriate role to the elements and look for them that way.